tools/code-generation/generator/src/main/java/com/amazonaws/util/awsclientgenerator/generators/cpp/s3/S3RestXmlCppClientGenerator.java

```diff
@@ -225,21 +225,21 @@ public SdkFileEntry[] generateSourceFiles(ServiceModel serviceModel) throws Exce
                 });
 
         serviceModel.getOperations().values().stream()
-                .filter(operationEntry -> operationEntry.getName().equals("WriteGetObjectResponse"))
+                .filter(operationEntry -> operationEntry != null && operationEntry.getName() != null && "WriteGetObjectResponse".equals(operationEntry.getName() ))
```

Comment:
I'm not sure why this piece of legacy code needs to be updated for the given feature.
Please provide some context.

Reply:
Initially I used S3 client to create the blob and found some null checks missing, so added it . But this is not necessary as with null operation name, we expect to fail